src/Compilers/Core/RebuildTest/DeterministicKeyBuilderTests.cs

```diff
@@ -312,6 +313,24 @@ JObject getValue(bool deterministic)
             }
         }
 
+        /// <summary>
+        /// Disabling determinism should mean all calls to GetDeteriministicKey return different values.
```

Comment:
It seems like it would be reasonable to just throw if GetDeterministicKey is called when determinism is disabled.

Reply:
I considered that but for now decided against it. It's not inherently wrong to ask for a key on a non-deterministic build, we just have to meet the contract that it never compares equal. I also worried that it would create friction points for consumers where they have to keep taking different paths whether a specific option was passed to the compiler. 

I actually started wondering if the opposite was needed. Essentially make the output deterministic even if the options weren't deterministic. Consider the C# IDE case as motivation for this. They are trying to determine equivalence between open / closing the IDE and remote sessions. That shouldn't all fall apart if the customer didn't use deterministic builds. It also seems odd to call `WithDeterminism(true)` before calling the API. Possible better to just ignore that option in the output.

Reply:
It seems like something we'll figure out the answer for as we set up the consuming side of the API. I agree there is a considerable use case for wanting to determine "equivalence" of compilations even if determinism is disabled. I could see that being its own option or just being the way it always works (if you don't care whether it's literally deterministic, isn't it good enough that the compilation inputs are equivalent except for the things which depend on the moment in time or are straight up random, etc.)